### PR TITLE
Reworking MQTT 5 Properties and integrating paho to RPC samples and code

### PR DESCRIPTION
**Pull request by RLeclair**
In this PR:
- Integration of Paho Async C with the RPC code. 
- Two new samples for Paho RPC.
- Re-work of MQTT property read and free. 

NOTE: Added topic filter to received topic matching function using az_span. This is temporary.

### CODE REVIEW

File: sdk/samples/core/mosquitto_rpc_client_sample.c [hunk 1]
```diff
@@ -45,6 +45,11 @@ static uint8_t correlation_id_buffers[RPC_CLIENT_MAX_PENDING_COMMANDS]
                                      [AZ_MQTT5_RPC_CORRELATION_ID_LENGTH];
 static pending_commands_array pending_commands;
 
+static az_mqtt5_options options = {
+  .certificate_authority_trusted_roots = AZ_SPAN_LITERAL_EMPTY,
+  .disable_tls_validation = true,
```
Comment (vaavva) [inline sdk/samples/core/mosquitto_rpc_client_sample.c:50]: Aren't we using TLS in this sample?

File: sdk/src/azure/platform/az_pahoasync5.c [hunk 2]
```diff
@@ -429,6 +429,7 @@ AZ_NODISCARD az_result az_mqtt5_property_bag_clear(az_mqtt5_property_bag* proper
 {
   _az_PRECONDITION_NOT_NULL(property_bag);
 
+  MQTTProperties_free(property_bag->pahoasync_properties);
   property_bag->pahoasync_properties->count = 0;
```
Comment (vaavva) [inline sdk/src/azure/platform/az_pahoasync5.c:433]: I think this line is now redundant - is this done within MQTTProperties_free?

**Review by vaavva — COMMENTED** [review]
will finish reviewing the test files tomorrow

File: sdk/samples/core/paho_multiple_rpc_sample.c [hunk 3]
```diff
@@ -0,0 +1,591 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file
+ * @brief RPC Server & Client sample for Mosquitto MQTT that supports multiple commands under the
```
Comment (vaavva) [inline sdk/samples/core/paho_multiple_rpc_sample.c:6]:
```suggestion
 * @brief RPC Server & Client sample for Paho MQTT that supports multiple commands under the
```

File: sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c [hunk 4]
```diff
@@ -359,114 +359,86 @@ static az_result subscribing(az_event_policy* me, az_event event)
...
+    az_mqtt5_rpc_client_rsp_event_data resp_data = { .response_payload = recv_data->payload,
                                                      .status = AZ_MQTT5_RPC_STATUS_UNKNOWN,
                                                      .error_message = AZ_SPAN_EMPTY,
                                                      .content_type = AZ_SPAN_EMPTY,
                                                      .correlation_id = AZ_SPAN_EMPTY };
 
-    az_result rc = _parse_response(
-        recv_data, &correlation_data, &status, &error_message, &content_type, &resp_data);
+    az_result rc = AZ_OK;
+    if (recv_data->properties == NULL)
```
[the remark this answers is not in the document]
Comment (RLeclair) [reply sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c:408]: Removed check

File: sdk/tests/core/test_az_mqtt5_rpc_client_hfsm.c [hunk 5]
```diff
@@ -880,6 +887,10 @@ static void test_az_mqtt5_rpc_client_invoke_begin_faulted_failure(void** state)
       az_mqtt5_rpc_client_invoke_begin(&test_rpc_client, NULL), AZ_ERROR_HFSM_INVALID_STATE);
 
   assert_int_equal(ref_pub_req, 0);
+
+#if defined(TRANSPORT_PAHO)
+  MQTTAsync_destroy(&test_client);
```
Comment (vaavva) [inline sdk/tests/core/test_az_mqtt5_rpc_client_hfsm.c:892]: Do we need `MQTTProperties_free` here too?

File: sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c [hunk 6]
```diff
@@ -359,114 +359,86 @@ static az_result subscribing(az_event_policy* me, az_event event)
...
+    {
+      resp_data.error_message = AZ_SPAN_FROM_STR("Response does not have properties.");
+      rc = AZ_ERROR_ITEM_NOT_FOUND;
+    }
+    else if (az_span_ptr(correlation_data_bindata) == NULL)
```
[the remark this answers is not in the document]
Comment (RLeclair) [reply sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c:413]: I changed the correlation_id and content type to be assigned directly when creating `resp_data` and changed the check to `az_span_is_content_equal`, thanks for pointing this out!

File: sdk/src/azure/platform/az_pahoasync5.c (hunk 2 above)
Comment (RLeclair) [reply sdk/src/azure/platform/az_pahoasync5.c:433]: It does zero out all fields, removed line.

**Review by vaavva — APPROVED** [review]

**Pull request merged**